webpack/ForemanTasks/Components/TaskDetails/TaskDetailsActions.js

```diff
@@ -0,0 +1,109 @@
+import API from 'foremanReact/API';
+import {
+  showLoading,
+  hideLoading,
+} from 'foremanReact/components/Layout/LayoutActions';
+import {
+  FOREMAN_TASK_DETAILS_FETCH_TASK_REQUEST,
+  FOREMAN_TASK_DETAILS_FETCH_TASK_SUCCESS,
+  FOREMAN_TASK_DETAILS_FETCH_TASK_FAILURE,
+  FOREMAN_TASK_DETAILS_STOP_POLLING,
+  FOREMAN_TASK_DETAILS_START_POLLING,
+  FOREMAN_TASK_DETAILS_TOGGLE_UNLOCK_MODAL,
+  FOREMAN_TASK_DETAILS_TOGGLE_FORCE_UNLOCK_MODAL,
+} from './TaskDetailsConstants';
+
+export const taskReloadStop = timeoutId => {
+  if (timeoutId) {
+    clearTimeout(timeoutId);
+  }
+  return {
+    type: FOREMAN_TASK_DETAILS_STOP_POLLING,
+    payload: { timeoutId },
+  };
+};
+
+export const taskReloadStart = (
+  timeoutId,
+  refetchTaskDetails,
+  id,
+  loading = false
+) => {
+  if (!timeoutId) {
+    timeoutId = setInterval(() => refetchTaskDetails(id, loading), 5000);
+  }
+  return {
+    type: FOREMAN_TASK_DETAILS_START_POLLING,
+    payload: { timeoutId },
+  };
+};
+
+export const refetchTaskDetails = (id, loading) => dispatch => {
+  if (!loading) {
+    showLoading();
+    dispatch(startRequest());
+    reloadTasksDetails(id, dispatch);
+  }
+};
+
+const reloadTasksDetails = async (id, dispatch) => {
+  try {
+    const { data } = await API.get(`/foreman_tasks/api/tasks/${id}/details`);
+    dispatch(requestSuccess(data));
+  } catch (error) {
+    dispatch(requestFailure(error));
+    document.location.reload();
+  } finally {
+    hideLoading();
+  }
+};
+
+export const fetchTaskDetails = (
+  id,
+  timeoutId,
+  refetchTaskDetailsAction
+) => dispatch => {
+  showLoading();
+  dispatch(startRequest());
+  getTasksDetails(id, dispatch, timeoutId, refetchTaskDetailsAction);
+};
+
+const getTasksDetails = async (
+  id,
+  dispatch,
+  timeoutId,
+  refetchTaskDetailsAction
+) => {
+  try {
+    const { data } = await API.get(`/foreman_tasks/api/tasks/${id}/details`);
+    dispatch(requestSuccess(data));
+    if (data.state !== 'stopped') {
```

Comment:
Currently clicking the "start auto reloading" when the task is stopped does only one fetch and never really enables the polling. If the user can't really enable the polling for a stopped task, shouldn't the button be disabled?

Reply:
This was done based on your previous comment:
> When you're looking at a stopped task and turn on auto reload, it keeps refreshing. Before this PR it would refresh once, figure out the task is already stopped and won't change in the future so it stopped reloading.

Did you mean something else?

Reply:
Sorry about that, I should have been more clear about it, the old behavior was a bit different.

The old way:
1) User clicks the button
2) Polling is enabled, button text changes
3) A single fetch is done
4) Frontend code determines the task cannot change in the future
5) Polling gets disabled, button text changes back

The new way is
1) User clicks the button
2) The frontend code determines the task cannot change in the future
3) A single fetch is done

Reply:
Thanks for clearing it up! fixed, after the request is successful if the state is still stopped, then polling is stopped. Is this the right behavior?

Reply:
Almost, currently:
1) User clicks the button
2) Polling is enabled, button text changes, `FOREMAN_TASK_DETAILS_START_POLLING` is dispatched
3) A single fetch is done
4) Frontend code determines the task cannot change in the future
5) Polling gets disabled, text button changes back

However if I click the button again, I'd expect it to start over. Instead, `FOREMAN_TASK_DETAILS_STOP_POLLING` is dispatched and nothing happens.

Reply:
Should work as expected now, thanks!